.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,47 @@
+{
+	"name": "kube-dev",
+
+	// TODO publish a version of this image to the organization's package registry and then comment
+	// out the `dockerFile` configuration.
+	//"image": "ghcr.io/kube-rs/dev:v1",
+	"dockerFile": "./Dockerfile",
+
+	// These extensions are loaded for all users by default.
+	"extensions": [
+		"DavidAnson.vscode-markdownlint",
+		"matklad.rust-analyzer",
+		"NathanRidley.autotrim",
+		"samverschueren.final-newline",
+		"streetsidesoftware.code-spell-checker",
+		"tamasfe.even-better-toml",
+		"vadimcn.vscode-lldb",
+		//"zxh404.vscode-proto3"
+	],
+
+	// Set project-specific VS Code settings here.
+	"settings": {
+		// Prevent conflicting versions of the Kubernetes API from being loaded.
+		"rust-analyzer.cargo.allFeatures": false,
+	},
+
+	// `make clippy` and `make lint` require the nightly toolchain, so install
+	// it as the container is created.
+	"postCreateCommand": "rustup update nightly",
+
+	// Configure docker.
+	"runArgs": [
+		"--init",
+		// Use the host network so we can access k3d, etc.
+		"--net=host",
+		// For lldb
+		"--cap-add=SYS_PTRACE",
+		"--security-opt=seccomp=unconfined"
+	],
+	"overrideCommand": false,
+	"remoteUser": "code",
+
+	// Mount the parent host's docker socket into the container.
+	"mounts": [
+		"source=/var/run/docker.sock,target=/var/run/docker-host.sock,type=bind"
+	]
```

Comment:
hah. i would have thought going through docker to run k3s in docker would have been one of those recursive exercises that would bring tons of caveats